app/src/main/java/deakin/gopher/guardian/model/Patient.kt

```diff
@@ -0,0 +1,91 @@
+package deakin.gopher.guardian.model;
+
+data class Patient(
```

Comment:
You don't need all the getters and setters. You also probably don't need the two constructors. And I doubt all the fields should be nullable. Most would likely be required

Reply:
If I remove all getters and setters, and also remove the constructors then this code will not work. Because whenever I make an object of the patient, I have to pass all the values. So I couldn't understand how is that possible. and I made values nullable because if I don't then the constructor gives an error. Can you give me a detailed comment on that?

Reply:
Okay if it's breaking code it's fair to leave the secondary constructors in. Improving the modelling is probably out of scope here so we can think about working on that later

Reply:
The build errors you get for the fields is because they are all set to private, which means they can't be accessed outside the class.

You can set them to public. Also the `Key` param can be removed from the first constructor, and some constructor params should be camelCase

```kotlin
package deakin.gopher.guardian.model

data class Patient(
    val patientId: String? = null,
    var address: String? = null,
    var dob: String? = null,
    var patientName: String? = null,
    var phone: String? = null,
    var photo: String? = null,
    var underCare: String? = null,
    var firstName: String? = null,
    var middleName: String? = null,
    var lastName: String? = null,
    var medicareNo: String? = null,
    var westwenAffairesNo: String? = null,
    var nokId1: String? = null,
    var nokId2: String? = null,
    var gpId1: String? = null,
    var gpId2: String? = null,
) {

    constructor(firstname: String, lastname: String) : this() {
        firstName = firstname
        lastName = lastname
    }

    constructor(
        dateOfBirth: String,
        firstname: String,
        middleName: String,
        lastname: String,
        medicareNo: String,
        westernAffairsNo: String
    ) : this() {
        dob = dateOfBirth
        firstName = firstname
        this.middleName = middleName
        lastName = lastname
        this.medicareNo = medicareNo
        westwenAffairesNo = westernAffairsNo
    }
}
```